### PR TITLE
Add `send_message()` function to the stdlib to expose the `SMO` opcode

### LINKED ISSUE #1868: Add `send_message()` function to the stdlib to expose the `SMO` opcode
The current work on the bridge project requires passing of `MessageOutput`s via the `SMO` opcode.
This functionality needs to be exposed in the stdlib.

reference: 
https://github.com/FuelLabs/fuel-specs/blob/bd6ec935e3d1797a192f731dadced3f121744d54/specs/protocol/tx_format.md#outputtype

https://github.com/FuelLabs/fuel-specs/blob/bd6ec935e3d1797a192f731dadced3f121744d54/specs/vm/instruction_set.md#smo-send-message-to-output

### PR DESCRIPTION
**Pull request by Braqzen**
## Changes

- Added send_message() to new module message.sw in `std`
- Added incomplete test to token_ops in sdk_harness

## Notes

- Test fails and I don't know why or how to fix it (apparently fuel-core does not support this yet and thus it panics)

## Related

Closes #1868

### CODE REVIEW

**Review by nfurfaro — COMMENTED** [review]
A few comments left. Once the `smo` opcode is fully functional and testing can be completed I'll take another pass.

File: sway-lib-std/src/token.sw [hunk 1]
```diff
@@ -113,3 +113,10 @@ pub fn transfer_to_output(amount: u64, asset_id: ContractId, to: Address) {
         };
     }
 }
+
+/// Sends message to `recipient` of length `msg_len` through `output` and amount of `coins`
+pub fn send_message(recipient: b256, msg_len: u64, output: u64, coins: u64) {
```
[the remark this answers is not in the document]
Comment (Braqzen) [reply sway-lib-std/src/token.sw:118]: According to the [instruction set](https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/instruction_set.md#smo-send-message-to-output) 
- `$rB` is "Length of message data, in bytes"
- `$rC` is some unspecified output as far as I can interpret (I'm leaning towards an index based on its usage there)

I can certainly move the function into a new module but I don't think I can enhance the documentation myself

[the remark this answers is not in the document]
Comment (nfurfaro) [reply sway-lib-std/src/token.sw:118]: >$rB is "Length of message data, in bytes"

Yep, so it is. ignore my earlier comment.

cc @adlerjohn on the `$rc` register for `smo`

[the remark this answers is not in the document]
Comment (adlerjohn) [reply sway-lib-std/src/token.sw]: Syd is correct that `$rc` is the index of the message output to use up.

**Comment by nfurfaro** [comment]
@Braqzen Are there any blockers here, or could this proceed now that `fuel-core` support is complete?

**Comment by nfurfaro** [comment]
Closing in favor of #2796

**Pull request closed without merging**